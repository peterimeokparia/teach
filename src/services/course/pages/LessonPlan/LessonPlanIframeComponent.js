import React from 'react'

import ReactIframe from '../Components/ReactIframe';



const LessonPlanIframeComponent = ({ className, name, source, width, height, allow, scrolling, frameBorder}) => {

  return(
        
          ( source ) ? <div>
                    <ReactIframe
                         className={className}
                         name={name }
                         source={source}
                         width={width}
                         height={height}
                         allow={allow}
                         scrolling={scrolling}
                         frameBorder={frameBorder}
                    />  
 
                  </div>

                 : <div></div>
        
     );
}

export default LessonPlanIframeComponent;