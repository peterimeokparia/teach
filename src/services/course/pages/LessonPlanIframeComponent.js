import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import ReactIframe from './ReactIframe';



const LessonPlanIframeComponent = ({ className, name, source, width, height, allow, scrolling, frameBorder}) => {

  return(
          <div>
               <ReactIframe

                        className={className}
                        name={name }
                        source={source}
                        width={width}
                        height={height}
                        allow={allow}
                        scrolling={scrolling}
                        frameBorder={frameBorder}
                />  
 
          </div>
     );
}

export default LessonPlanIframeComponent;