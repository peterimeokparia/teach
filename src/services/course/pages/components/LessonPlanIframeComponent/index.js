import ReactIframe from 'services/course/pages/components/ReactIframe';

const LessonPlanIframeComponent = ({
     className,
     id, 
     name, 
     source, 
     width,  
     height, 
     allow, 
     scrolling, 
     frameBorder }) => {
return(( source ) 
     ? <div id={id}>
          <ReactIframe
               className={className}
               name={name}
               source={source}
               width={width}
               height={height}
               allow={allow}
               scrolling={scrolling}
               frameBorder={frameBorder}
          />  
          </div>
     : <div></div>

     );
};

export default LessonPlanIframeComponent;