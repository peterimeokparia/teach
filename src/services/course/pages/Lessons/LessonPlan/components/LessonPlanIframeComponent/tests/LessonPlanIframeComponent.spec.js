import React from 'react';
import LessonPlanIframeComponent  from 'services/course/pages/Lessons/LessonPlan/components/LessonPlanIframeComponentent/tests/node_modules/services/course/pages/Lessons/LessonPlan_test/components/LessonPlanIframeComponent';
import renderer from 'react-test-renderer';

describe('LessonPlanIframeComponent', () =>  {  
 
  it('renders as expected', async () => {

    const tree = renderer.create(
        <LessonPlanIframeComponent 
            name="embed_readwrite" 
            source={"test"}
            width={"1536px"}
            height="900px"
            allow="camera;microphone"
            scrolling="yes"
            frameBorder="0" 
      /> 
    );

      expect(tree.toJSON()).toMatchSnapshot();
  })
});




