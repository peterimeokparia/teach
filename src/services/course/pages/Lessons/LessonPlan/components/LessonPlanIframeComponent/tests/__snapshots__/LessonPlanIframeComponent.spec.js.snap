// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LessonPlanIframeComponent renders as expected 1`] = `
<div>
  <div
    className="iframe"
  >
    <iframe
      allow="camera;microphone"
      frameBorder="0"
      height="900px"
      name="embed_readwrite"
      scrolling="yes"
      src="test"
      title="embed_readwrite"
      width="1536px"
    />
  </div>
</div>
`;
